Kitodo/src/main/java/org/kitodo/production/forms/ProzesskopieForm.java

```diff
@@ -498,7 +511,31 @@ protected void removeCollectionsForChildren(LegacyMetsModsDigitalDocumentHelper
      * Create the process and save the metadata.
      */
     public String createNewProcess() {
+        if (Objects.nonNull(titleRecordLinkTab.getTitleRecordProcess())) {
+            if (Objects.isNull(titleRecordLinkTab.getInsertionPositionSelectionSelectedItem())) {
+                Helper.setErrorMessage("prozesskopieForm.createNewProcess.noInsertionPositionSelected");
+                return stayOnCurrentPage;
+            } else {
+                User titleRecordOpenUser = DataEditorForm
+                        .getUserOpened(titleRecordLinkTab.getTitleRecordProcess().getId());
+                if (Objects.nonNull(titleRecordOpenUser)) {
+                    Helper.setErrorMessage("prozesskopieForm.createNewProcess.titleRecordOpen",
+                        titleRecordOpenUser.getFullName());
+                    return stayOnCurrentPage;
+                }
+            }
+        }
         if (createProcess()) {
+            if (Objects.nonNull(titleRecordLinkTab.getTitleRecordProcess())) {
+                ServiceManager.getProcessService().refresh(prozessKopie);
+                try {
+                    MetadataEditor.addLink(titleRecordLinkTab.getTitleRecordProcess(),
```

Comment:
The `MetadataEditor` class contains functions that manipulate the metadata in a programmatic way. These include adding the links or assigning the pagination. I have separated the editing functions from the form classes, because forms should operate the application logic, but should not implement them.

Reply:
Maybe the class `MetadataEditor` is a misnomer and should better be called `MetadataService`. But if so, we should do this in a separate pull request.